projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-trait.php

```diff
@@ -567,4 +567,55 @@ public function post_contains_text( $post_id, $text ) {
 
 		return ( '0' !== $result ) ? 1 : 0;
 	}
+
+	/**
+	 * Get additional fields data for Checkout and Post-Checkout events.
+	 *
+	 * @param WC_Order|null $order The order object or null if we don't have an order.
+	 *
+	 * return array An array containing the additional fields data.
+	 */
+	private function get_additional_fields_data( $order = null ) {
+		$data = array();
+
+		if ( class_exists( 'Automattic\WooCommerce\Blocks\Package' ) && class_exists( 'Automattic\WooCommerce\Blocks\Domain\Services\CheckoutFields' ) ) {
+			$additional_fields_controller = Automattic\WooCommerce\Blocks\Package::container()->get( Automattic\WooCommerce\Blocks\Domain\Services\CheckoutFields::class );
+			$additional_fields            = $additional_fields_controller->get_additional_fields();
+			$fields_count                 = count( $additional_fields );
+			$fields_data                  = array_map(
+				function ( $field_key, $field ) use ( $additional_fields_controller ) {
+					return array(
+						$field_key,
+						$additional_fields_controller->get_field_location( $field_key ),
+						$field['type'],
+						$field['required'] ? '1' : '0',
+					);
+				},
```

Comment:
opted here for a flat array with no keys to simplify querying it.

Reply:
Do we know that this really does simplify querying it? Have we checked with data teams about this?

Reply:
I have not, but accessing array index is easier than object values, in python at least.

Reply:
Yeah I agree on that but now  someone has to remember/document what order these fields are in, is that fine/usual in data? I'm not sure which is why I'm asking, if that's the way things work normally then I'm fine with it